lib/app/lifecycle.coffee

```diff
@@ -37,6 +38,7 @@ crypto = require 'crypto'
     .post(opts.ARTSY_URL + '/api/v1/user')
     .set({
       'X-Xapp-Token': artsyXapp.token,
+      'X-Forwarded-For': forwardedFor(req),
```

Comment:
One thing that's slightly different here and below (vs. the implementation in Force), is that this header being set is gated behind `if req?.connection?.remoteAddress?`.

Not sure if you need that (looking in https://github.com/artsy/force/blob/ec3343bbbf07c4e2d6c0a5bf272d06274744b017/src/lib/metaphysics2.coffee#L33)

Reply:
I assumed that was just for testing. As far as I can tell, [real] express requests always have a `connection`.

Reply:
Perhaps the `remoteAddress` part is `null` or `undefined`? (and thus, this guard prevents _falsely_ setting this header).

I suppose based on our current infrastructure however, it _should_ always exist in a deployed environment (right)?

Reply:
Dug some more... express's `req` is "an enhanced version of Node’s own request object and supports all [built-in fields and methods](https://nodejs.org/api/http.html#http_class_http_incomingmessage)." So I think deviations are only possible in stubbed environments.